### DISCUSSION TITLE
Edge-config get vs getAll

### DISCUSSION THREAD

**Discussion by willemliufdmg** [vercel/community#2434] [Help] [answered]
### Problem Description

Hi, I can't find it in the documentation but the Vercel pricing states that costs are based on the number of Edge Config reads.

Does each call to the `get` function of the EdgeConfigClient (`@vercel/edge-config`) count as 1 read? And does each subsequent call to `get` count as an extra read as well? Or does the SDK do some caching and are all calls made within the same request in the same file (e.g. Middleware) count as 1?

Assuming that each `get` does count as a separate read. If I do a `getAll` and pass along an array with `keys`. Does this count as a single read? Or does each `key` count as a read?

**Comment by dferber90** [comment, marked as answer]
Apologies for the late response, I just found this discussion by chance.

- Each `get` call counts as one read
- A `getAll(['keyA', 'keyB'])` call with multiple keys counts as a single read as those reads will be batched
- A `getAll()` call with no arguments (reading the whole edge config) counts as a single read
